tests/slsa_analyzer/git_service/test_gitlab.py

```diff
@@ -41,7 +44,70 @@ def test_construct_clone_url_without_token(repo_url: str) -> None:
 )
 def test_construct_clone_url_with_token(repo_url: str, clone_url: str) -> None:
     """Test if the ``construct_clone_url`` method produces proper clone URLs with the access token."""
-    with mock.patch.dict(os.environ, {"MCN_PUBLIC_GITLAB_TOKEN": "abcxyz"}):
-        gitlab = PublicGitLab()
+    with mock.patch.dict(os.environ, {"MCN_GITLAB_TOKEN": "abcxyz"}):
+        gitlab = PubliclyHostedGitLab()
         gitlab.load_defaults()
         assert gitlab.construct_clone_url(repo_url) == clone_url
+
+
+@pytest.mark.parametrize(
+    ("user_config_input", "repo_url", "clone_url"),
+    [
+        pytest.param(
+            """
+            [git_service.gitlab.self_hosted]
+            domain = internal.gitlab.org
+            """,
+            "https://internal.gitlab.org/owner/repo.git",
+            "https://oauth2:abcxyz@internal.gitlab.org/owner/repo.git",
+            id="Self-hosted GitLab is set in user config.",
+        ),
+        pytest.param(
+            """
+            [git_service.gitlab.self_hosted]
+            domain = internal.gitlab.org
+            """,
+            "https://internal.gitlab.org/owner/repo",
+            "https://oauth2:abcxyz@internal.gitlab.org/owner/repo",
+            id="Self-hosted GitLab is set in user config.",
```

Comment:
Not sure if this `id` is useful because it's the same as the previous parameter `id`.

Reply:
You are right. I will remove it.

Reply:
Removed in https://github.com/oracle/macaron/pull/352/commits/0f6ee7933caa7609670f73c4cf40d149c35b2f5a